frontend/src/pages/notebookController/NotebookControlPanelRedirect.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import ApplicationsPage from 'pages/ApplicationsPage';
+import { getUserStateFromDashboardConfig } from 'utilities/notebookControllerUtils';
+import AppContext from '../../app/AppContext';
+import NotebookControllerContext from './NotebookControllerContext';
+
+const NotebookControlPanelRedirect: React.FC = () => {
+  const history = useHistory();
+  const { username: translatedUsername } = useParams<{ username: string }>();
+  const { dashboardConfig } = React.useContext(AppContext);
+  const { setCurrentUserState } = React.useContext(NotebookControllerContext);
+  React.useEffect(() => {
+    if (translatedUsername && dashboardConfig.spec.notebookController?.enabled) {
+      const userState = getUserStateFromDashboardConfig(
+        translatedUsername,
+        dashboardConfig.status?.notebookControllerState,
```

Comment:
```suggestion
        dashboardConfig.status.notebookControllerState,
```

Reply:
I don't think we could remove the `?` here. The status could be undefined if there is no `notebookControllerState` at all like you said (If someone manually changes the dashboard config on the cluster)

Reply:
Ah right you are... ignore please 🙂